File: python/1a.py
# so basically a for loop and you go - or + but then loop back around
import math

pos = 50
count = 0

with open("C:/Users/saanv/Downloads/input.txt", "r") as lines:
  for line in lines:
    val = int(line[1:])
    if line[0] == "L":
      pos = (pos - val) % 100
    else:
      pos = (pos + val) % 100
      
    if pos == 0:
      count += 1

print(count)

# answer 1: 16, but not right
# answer 2: 307, fixed logic on rotating right, but still too low?? the test case works tho so idk what i got wrong, it has to be smth with the nuance of 99 or 0
# answer 3: 1172, now I accounted for line values being above 100 thanks to the subreddit, it's correct yay!!!

File: java/1a.java
// this actually needs to be named a1.java, but I like having my files nice and neat
import java.io.*;
import java.util.*;

public class a1 {
  public static void main(String[] args) throws FileNotFoundException {
    int pos = 50;
    int count = 0;
    
    InputStream lines = new FileInputStream("C:/Users/saanv/Downloads/input.txt");
    Scanner input = new Scanner(lines);
    while (input.hasNextLine()) {
      String next = input.nextLine();
      int val = Integer.parseInt(next.substring(1)) % 100;

      if (next.substring(0, 1).equals("L")) {
        if (pos - val < 0) {
          pos = 100 - Math.abs(pos - val);
        } else {
          pos -= val;
        }
      } else {
        if (pos + val > 99) {
          pos = 0 + Math.abs(100 - (pos + val));
        } else {
          pos += val;
        }
      }

      if (pos == 0) {
        count++;
      }
      
    }
    System.out.print(count);
  }
}
